src/librustc/ich/impls_ty.rs

```diff
@@ -1133,7 +1133,7 @@ impl<'a, 'gcx> HashStable<StableHashingContext<'a>> for ty::CratePredicatesMap<'
 
 impl_stable_hash_for!(struct ty::AssociatedItem {
     def_id,
-    name,
+    ident,
```

Comment:
@michaelwoerister 
A suspicious detail: for `ty::FieldDef` I had to hash only `ident.name` instead of full `ident`, otherwise I got some linking errors while building standard library crates, but for `ty::AssociatedItem` hashing the full `ident` was still okay for some reason.